libbeat/outputs/kafka/partition.go

```diff
@@ -272,11 +272,11 @@ func makeFieldsHashPartitioner(log *logp.Logger, fields []string, dropFail bool)
 }
 
 func hash2Partition(hash uint32, numPartitions int32) (int32, error) {
-	p := int32(hash)
+	p := int32(hash) % numPartitions
```

Comment:
how about 
```
p := hash % numPartitions
return p, nil
```
?

Reply:
gives compiler error

./partition.go:275:12: invalid operation: hash % numPartitions (mismatched types uint32 and int32)